qa/pull-tester/rpc-tests.py

```diff
@@ -184,26 +182,30 @@
     'replace-by-fee.py',
 ]
 
-
 def runtests():
-    test_list = []
-    if '-extended' in opts:
-        test_list = testScripts + testScriptsExt
-    elif len(opts) == 0 or (len(opts) == 1 and "-win" in opts):
-        test_list = testScripts
+    # Build list of tests
+    test_scripts = BASE_SCRIPTS
+    if ENABLE_ZMQ:
+        test_scripts += ZMQ_SCRIPTS
+    if args.extended:
+        test_scripts += EXTENDED_SCRIPTS
+
+    if len(tests) == 0:
+        test_list = test_scripts
     else:
-        for t in testScripts + testScriptsExt:
-            if t in opts or re.sub(".py$", "", t) in opts:
+        test_list = []
+        for t in test_scripts:
```

Comment:
One shouldn't be required to pass --extended to be able to run specific tests from the extended set. You can leave the following here:
```
for t in BASE_SCRIPTS + EXTENDED_SCRIPTS:
```
Mhmm, that would still exclude zmq_test.py because of the other change you made...